core/runtime/op_driver/mod.rs

```diff
@@ -9,31 +8,66 @@ use std::task::Poll;
 
 mod erased_future;
 mod future_arena;
+mod futures_unordered_driver;
 mod joinset_driver;
 mod op_results;
 
+#[allow(unused)]
+pub use futures_unordered_driver::FuturesUnorderedDriver;
+#[allow(unused)]
 pub use joinset_driver::JoinSetDriver;
 
-pub type RetValMapper<R> =
-  for<'r> fn(
-    &mut v8::HandleScope<'r>,
-    R,
-  ) -> Result<v8::Local<'r, v8::Value>, serde_v8::Error>;
+pub use self::op_results::OpMappingContext;
+pub use self::op_results::OpResult;
+pub use self::op_results::V8OpMappingContext;
+pub use self::op_results::V8RetValMapper;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum OpScheduling {
+  Eager,
+  Lazy,
+  Deferred,
+}
 
 /// `OpDriver` encapsulates the interface for handling operations within Deno's runtime.
 ///
 /// This trait defines methods for submitting ops and polling readiness inside of the
 /// event loop.
-pub(crate) trait OpDriver: Default {
+///
+/// The driver takes an optional [`OpMappingContext`] implementation, which defaults to
+/// one compatible with v8. This is used solely for testing purposes.
+pub(crate) trait OpDriver<C: OpMappingContext = V8OpMappingContext>:
```

Comment:
I think that complicates code a lot - in a follow up let's figure out another way to test this so we don't have to pass `OpMappingContext` everywhere.

Reply:
It would be great to remove this somehow, but I haven't figured out a way to do this yet. What really kills us is that this code needs to run under Miri :/